### PR TITLE
CEDS-2033 Split totalItemInformation page

### PR DESCRIPTION
**Pull request by bambuchaAdm**

### CODE REVIEW

File: app/controllers/declaration/TotalNumberOfItemsController.scala [hunk 1]
```diff
@@ -60,12 +60,7 @@ class TotalNumberOfItemsController @Inject()(
   }
 
   private def nextPage(declarationType: DeclarationType): Mode => Call =
-    declarationType match {
-      case DeclarationType.SUPPLEMENTARY | DeclarationType.STANDARD | DeclarationType.CLEARANCE =>
-        controllers.declaration.routes.NatureOfTransactionController.displayPage
-      case DeclarationType.SIMPLIFIED | DeclarationType.OCCASIONAL =>
-        controllers.declaration.routes.PreviousDocumentsController.displayPage
-    }
+    controllers.declaration.routes.TotalPackageQuantityController.displayPage
```
Comment (PatrykRudnicki) [inline app/controllers/declaration/TotalNumberOfItemsController.scala]: If now it's just one line without logic, what do you think about inlining this?

File: app/models/requests/AuthenticatedRequest.scala [hunk 2]
```diff
@@ -19,6 +19,6 @@ package models.requests
 import models.SignedInUser
 import play.api.mvc.{Request, WrappedRequest}
 
-class AuthenticatedRequest[A](request: Request[A], val user: SignedInUser) extends WrappedRequest[A](request) {
+class AuthenticatedRequest[+A](request: Request[A], val user: SignedInUser) extends WrappedRequest[A](request) {
```
Comment (PatrykRudnicki) [inline app/models/requests/AuthenticatedRequest.scala]: Is this necessary? How we need it now?

File: test/views/declaration/TotalPackageQuantityViewSpec.scala [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          view.getElementById("section-header").text() must include("supplementary.items")
+        }
+
+        "display header" in {
+          view.getElementById("title").text() must be("supplementary.valueOfItems")
+        }
+
+        "display empty input with label for Total Package" in {
+          view.getElementById("totalPackage-label").text() must be("supplementary.totalPackageQuantity") // FIXME
+//          view.getElementById("totalPackage-hint").text() must be("supplementary.totalPackageQuantity.hint") //FIXME there is no hint
```
Comment (PatrykRudnicki) [inline test/views/declaration/TotalPackageQuantityViewSpec.scala]: If there is no hint please remove this commented code

File: app/controllers/declaration/TotalNumberOfItemsController.scala (hunk 1 above)
Comment (bambuchaAdm) [reply app/controllers/declaration/TotalNumberOfItemsController.scala]: In next PR about skiping - it won't be one line

File: app/models/requests/AuthenticatedRequest.scala (hunk 2 above)
Comment (bambuchaAdm) [reply app/models/requests/AuthenticatedRequest.scala]: Technically wrappers around request should have same variance as orginal Request so it should be covariant

File: test/views/declaration/TotalPackageQuantityViewSpec.scala (hunk 3 above)
Comment (bambuchaAdm) [reply test/views/declaration/TotalPackageQuantityViewSpec.scala]: Ok. This is rememberence for asking designers about copy here.

**Pull request merged**